march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -33,16 +35,29 @@ MarchHardwareInterface::~MarchHardwareInterface() = default;
 void MarchHardwareInterface::init()
 {
   // Start ethercat cycle in the hardware
-  this->marchRobot.startEtherCAT();
-
-  if (!this->marchRobot.isEthercatOperational())
+//  this->marchRobot.startEtherCAT();
+//
+//  if (!this->marchRobot.isEthercatOperational())
+//  {
+//    ROS_FATAL("EtherCAT is not operational");
+//    exit(0);
+//  }
+
+  urdf::Model model;
+  if (!model.initParam("/robot_description"))
   {
-    ROS_FATAL("EtherCAT is not operational");
-    exit(0);
+      ROS_ERROR("Failed to read the urdf from the parameter server.");
+      throw std::runtime_error("Failed to read the urdf from the parameter server.");
   }
 
-  // Get joint names
-  nh_.getParam("/march/hardware_interface/joints", joint_names_);
+  // Get joint names from urdf
+  for (auto const& urdfJoint : model.joints_)
+  {
+      if (urdfJoint.second->type != urdf::Joint::FIXED)
+      {
+          joint_names_.push_back(urdfJoint.first);
```

Comment:
what if a joint is `allowActuation: true`, but the joint is fixed? This means the joint cannot move but high voltage is set to on for that joint?